src/tests.rs

```diff
@@ -37,6 +62,166 @@ fn can_deposit() {
         });
 }
 
+#[test]
+fn can_withdraw() {
+    MockGenesisConfig::with_balances(vec![(BOB, 500)])
+        .build()
+        .execute_with(|| {
+            assert_ok!(Bank::register_role(&ALICE, Role::Manager));
+            assert_eq!(
+                Accounts::<Runtime>::get(&BOB),
+                AccountData {
+                    free: 500,
+                    reserved: 0
+                }
+            );
+            System::reset_events();
+            assert_ok!(Bank::withdraw(RuntimeOrigin::signed(ALICE), BOB, 100));
+            assert_eq!(
+                System::events()[0].event,
+                RuntimeEvent::Bank(Event::<Runtime>::Withdraw {
+                    user: BOB,
+                    amount: 100
+                })
+            );
+            assert_eq!(
+                Accounts::<Runtime>::get(&BOB),
+                AccountData {
+                    free: 400,
+                    reserved: 0
+                }
+            );
+            assert_noop!(
+                Bank::withdraw(RuntimeOrigin::signed(ALICE), BOB, 500),
+                Error::<Runtime>::InsufficientBalance
+            );
+            assert_noop!(
+                Bank::withdraw(RuntimeOrigin::signed(ALICE), ALICE, 500),
+                Error::<Runtime>::IncorrectRole
+            );
+            assert!(Bank::check_total_issuance());
+        });
+}
+
+#[test]
+fn can_transfer() {
+    MockGenesisConfig::with_balances(vec![(ALICE, 1_000), (BOB, 500)])
+        .build()
+        .execute_with(|| {
+            assert_eq!(
```

Comment:
These 2 balance checks are not needed. The initial balance can be seen from the "with_balances" call